src/mixed-line-bar-chart/chart-popover.tsx

```diff
@@ -65,7 +68,26 @@ function MixedChartPopover<T extends ChartDataTypes>(
               onMouseLeave={onMouseLeave}
               onBlur={onBlur}
             >
-              <ChartSeriesDetails details={highlightDetails.details} />
+              <ChartSeriesDetails
+                key={highlightDetails.position}
+                details={highlightDetails.details}
+                setPopoverText={setPopoverText}
+                expandedSeries={expandedSeries[highlightDetails.position]}
+                setExpandedState={(index, isExpanded) =>
+                  setExpandedSeries(oldState => {
```

Comment:
Why are there no tests?

Reply:
Thanks, same as above, handled in [3c067a0](https://github.com/cloudscape-design/components/pull/1644/commits/3c067a09fbaca7e4e237a5c75e419b75cd4b3138)

Reply:
Added unit test coverage [here](https://github.com/cloudscape-design/components/pull/1644/files#diff-cf210ece79a3b18b604dcbb61742c9c3daaa12afd61ac6d8720af4b917efcb4aR105).

Also covered with integration tests including hover behavior [here](https://github.com/cloudscape-design/components/pull/1644/files#diff-8193a37686e2c08dc27e3187fa2e6d21748c49c8412f28e7506fd808e06fe69aR41).

Reply:
If I get this correctly, the expandedSeries keeps expanded states per X coordinate and series, like:

```
{
  'x=1': Set(0),
  'x=2': Set(),
  'x=3': Set(1) 
}
```

In the above example, the x=1, x=2, and x=3 are visited X coordinates and 0 and 1 are expanded series. Series 0 is expanded for x=1 only and series 1 is expanded for x=3 only.

I have two questions therefore:
1. Why do we decide to keep expanded state per x coordinate at all? When I drill-down on different coordinates it might not be very intuitive as of why sometimes a series is expanded.
2. What happens if the data is refreshed after the visit? The X coordinates are less concerning because those are tracked by formatted values. However, the series are tracked by indices. If the series order or structure change the state might not longer stay valid. This can easily happen if the series are filtered out with the default filter.

Reply:
Thanks for the detailed feedback. To your questions:
1. This was the outcome after discussing with design. Personally I don't have a strong opinion.
2. Changed to use series title as discussed offline.